torch/csrc/jit/script/class_type.cpp

```diff
@@ -9,19 +9,21 @@ namespace c10 {
 
 Function* ClassType::getMethod(const std::string& name) const {
   const auto qualname = QualifiedName(*qualified_name_obj(), name);
-  return compilation_unit_->find_function(qualname);
+  auto cu = compilation_unit_.lock();
+  TORCH_INTERNAL_ASSERT(cu);
+  return cu->find_function(qualname);
 }
 
-std::shared_ptr<CompilationUnit> ClassType::compilation_unit() {
+std::weak_ptr<CompilationUnit> ClassType::compilation_unit() {
```

Comment:
I think this should return a strong pointer via lock(). weak_ptr is an implementation detail of the class.